src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceBeforeCheck.java

```diff
@@ -209,19 +210,15 @@
 
     @Override
     public void visitToken(DetailAST ast) {
-        final String line = getLine(ast.getLineNo() - 1);
+        final int[] line = getLineCodePoints(ast.getLineNo() - 1);
         final int before = ast.getColumnNo() - 1;
-        final int[] codePoints = line.codePoints().toArray();
 
-        if ((before == -1 || CommonUtil.isCodePointWhitespace(codePoints, before))
+        if ((before == -1 || CommonUtil.isCodePointWhitespace(line, before))
                 && !isInEmptyForInitializerOrCondition(ast)) {
             boolean flag = !allowLineBreaks;
             // verify all characters before '.' are whitespace
-            for (int i = 0; i <= before - 1; i++) {
-                if (!CommonUtil.isCodePointWhitespace(codePoints, i)) {
-                    flag = true;
-                    break;
-                }
+            if (before > -1 && !CodePointUtil.hasWhitespaceBefore(before, line)) {
```

Comment:
Please separate update to this check into two commits; for the first one, do only update to use codepoint utils and `getLineCodePoints` method. For the second one, please refactor logic in this block to just have one descriptively named boolean variable comprised of all conditions that we place on `flag` variable now (`!allowLineBreaks && before > -1 && !CodePointUtil.hasWhitespaceBefore(before, line)`). Do you think something like `hasNoWhitespaceBefore` is a good name?

Reply:
After this update, please regenerate report.

Reply:
Resolved.

Reply:
I see now that we made a mistake in naming logic, this is not what this condition tells us. Let's rename to `isNotFirstTokenOnLine`

Reply:
done

Reply:
done

Reply:
<img width="596" alt="Screenshot 2022-03-15 at 9 48 40 PM" src="https://user-images.githubusercontent.com/79860504/158423759-fafbeb06-5433-4e74-9ac9-6f947cbf5d65.png">

Reply:
I suspected this may be a problem, let's just add this condition into `if` statement guarding log method then.

Reply:
need to combine all `if` condition

Reply:
done

Reply:
@nick-mancuso ping

Reply:
> need to combine all if condition as PMD was failing

I think resulting condition is hard to understand now, it looks like we should go back to variable approach, if nothing else we can name it `isViolation`.

Reply:
> need to combine all if condition as PMD was failing

I think resulting condition is hard to understand now, it looks like we should go back to variable approach, if nothing else we can name it `isViolation`.

Reply:
done

Reply:
Resolved.